src/Components/WebAssembly/testassets/Wasm.Authentication.Client/Pages/FetchData.razor

```diff
@@ -42,19 +41,18 @@ else
 
     protected override async Task OnInitializedAsync()
     {
-        var httpClient = new HttpClient();
-        httpClient.BaseAddress = new Uri(Navigation.BaseUri);
-
-        var tokenResult = await AuthenticationService.RequestAccessToken();
-
-        if (tokenResult.TryGetToken(out var token))
+        try
         {
-            httpClient.DefaultRequestHeaders.Add("Authorization", $"Bearer {token.Value}");
-            forecasts = await httpClient.GetFromJsonAsync<WeatherForecast[]>("WeatherForecast");
+            forecasts = await WeatherForecast.GetForecastAsync();
         }
-        else
+        catch (AccessTokenNotAvailableException exception)
         {
-            Navigation.NavigateTo(tokenResult.RedirectUrl);
+            exception.RedirectToLogin();
         }
     }
+
+    public void Dispose()
+    {
+        WeatherForecast.Dispose();
```

Comment:
I'm guessing that typed clients don't strictly have to be disposed, and disposing them is just a way to ensure their requests are cancelled if still in mid-flight. Is that right?

Reply:
Yep